crates/cext/src/dag.rs

```diff
@@ -1243,6 +1244,153 @@ pub unsafe extern "C" fn qk_dag_get_instruction(
     unsafe { instruction.write(inst) };
 }
 
+/// @ingroup QkDag
+/// Compose the ``other`` DAG onto the ``dag`` instance with the option of a subset
+/// of input wires of ``other`` being mapped onto a subset of output wires of ``dag``.
+///
+/// ``other`` may include a smaller or equal number of wires for each type.
+///
+/// @param dag A pointer to be composed on.
+/// @param other A pointer to the DAG to compose with ``dag``.
+/// @param qubits A list of indices representing the qubit wires to compose
+///     onto.
+/// @param num_qubits The number of qubits in ``qubits``.
```

Comment:
Don't we know the number of qubits and clbits from the `other` DAG already and can remove these arguments? We should then only add as safety comment that the `qubits` and `clbits` pointers should be readable for the appropriate number of elements.